autopush/logging.py

```diff
@@ -108,6 +105,23 @@ def __call__(self, event):
             self._output.write(unicode(text))
             self._output.flush()
 
+    def raven_log(self, event):
+        f = event["log_failure"]
+        stack = extra = None
+        tb = f.getTracebackObject()
+        if not tb:
+            # include the current stack for at least some context
+            stack = list(iter_stack_frames())[4:]  # approx.
```

Comment:
iter_stack_frames isn't "officially" public but I see it being used enough elsewhere that I'd consider it public enough (so likely stable enough). it's very simple, get the current stack in its particular format. raven strips frames from its result in a couple different ways so I expect it to not change.

I'll note 4: is more about the number of twisted logging frames (+ PushLogger's) involved, which could change too. 5 cleans it all up but I approximated to 4 just in case yet still getting rid of most of it